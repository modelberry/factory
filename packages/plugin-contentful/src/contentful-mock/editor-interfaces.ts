export const editorInterfaces: Record<string, any> = {
  testTopic: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 11,
      createdAt: '2021-08-06T16:53:45.223Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-08T19:29:00.057Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'testTopic', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'abstract',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'actions',
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'heading',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      { fieldId: 'icon', widgetId: 'singleLine', widgetNamespace: 'builtin' },
      {
        fieldId: 'media',
        widgetId: 'assetLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'poster',
        widgetId: 'assetLinkEditor',
        widgetNamespace: 'builtin',
      },
    ],
  },
  testAction: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 36,
      createdAt: '2021-08-05T21:24:24.051Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-08T19:28:57.798Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'testAction', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'anchor',
        settings: { helpText: 'Voorbeeld: mijn-anchor' },
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'description',
        settings: { helpText: 'Omschrijving van de actie, voor zoekmachines' },
        widgetNamespace: 'builtin',
      },
    ],
  },
  navigationSection: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:23.695Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:27.486Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'navigationSection', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      { fieldId: 'title', widgetId: 'singleLine', widgetNamespace: 'builtin' },
      {
        fieldId: 'logo',
        widgetId: 'assetLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'logoAction',
        widgetId: 'entryLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'header',
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'footer',
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'social',
        widgetId: 'entryLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'actions',
        widgetId: 'entryLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'legal',
        widgetId: 'entryLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'information',
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'sitemap',
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
    ],
  },
  topicSection: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:23.079Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:26.390Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'topicSection', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'variant',
        settings: { helpText: 'Hoe de sectie wordt weergegeven' },
        widgetId: 'dropdown',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'topics',
        settings: {
          helpText:
            'E\u00E9n of meerder onderwerpen die worden weergegeven in de sectie',
        },
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      { fieldId: 'hideIcon', widgetId: 'boolean', widgetNamespace: 'builtin' },
      { fieldId: 'hideMedia', widgetId: 'boolean', widgetNamespace: 'builtin' },
      {
        fieldId: 'hideHeading',
        widgetId: 'boolean',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'hideAbstract',
        widgetId: 'boolean',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'hideAction',
        widgetId: 'boolean',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'reversedOrder',
        widgetId: 'boolean',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'eventId',
        settings: { helpText: 'Voorbeeld: mijn-event-id' },
        widgetId: 'slugEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'theme',
        settings: { helpText: 'Kies het thema voor deze sectie' },
        widgetId: 'dropdown',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'modelVersion',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
    ],
  },
  topic: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:22.729Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:26.396Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'topic', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'heading',
        settings: { helpText: 'Titel van het onderwerp, zonder punt aan eind' },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'abstract',
        settings: {
          helpText:
            'Korte omschrijving van het onderwerp, sluit af met een punt',
        },
        widgetId: 'multipleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'media',
        settings: { helpText: 'Afbeelding bij het onderwerp' },
        widgetId: 'assetGalleryEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'icon',
        settings: { helpText: 'Icoon bij het onderwerp' },
        widgetId: 'dropdown',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'actions',
        settings: {
          helpText:
            'E\u00E9n of meerdere acties bij het onderwerp. Een actie heeft een eigen naam en kan ook linken naar een extern bron.',
        },
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'poster',
        settings: { helpText: '' },
        widgetId: 'assetLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'mediaEmbed',
        settings: {
          helpText:
            'Voeg Youtube/Vimeo embed code toe. Note: Media embed ondersteund alleen Embed-type: html',
        },
        widgetId: 'entryLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'modelVersion',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
    ],
  },
  textSection: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:22.396Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:26.393Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'textSection', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'variant',
        settings: { helpText: '' },
        widgetId: 'dropdown',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'text',
        settings: {
          helpText: 'Eenvoudig opgemaakte tekstveld met kopjes en afbeeldingen',
        },
        widgetId: 'richTextEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'eventId',
        settings: { helpText: 'Voorbeeld: mijn-event-id' },
        widgetId: 'slugEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'modelVersion',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
    ],
  },
  navigationSegment: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:21.871Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:25.398Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'navigationSegment', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
      },
      {
        fieldId: 'heading',
        settings: {
          helpText:
            'De naam van dit navigatiesegment. Alleen nodig bij meerdere segmenten.',
        },
        widgetId: 'singleLine',
      },
      {
        fieldId: 'abstract',
        settings: { helpText: 'Tekst voor dit menu segment' },
        widgetId: 'multipleLine',
      },
      {
        fieldId: 'actions',
        settings: { helpText: "De pagina's in dit navigatie segment." },
        widgetId: 'entryLinksEditor',
      },
      { fieldId: 'modelVersion' },
    ],
  },
  page: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:21.508Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:26.395Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'page', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'path',
        settings: {
          helpText:
            'Laatste deel van de URL naar deze pagina. Bijvoorbeeld: www.mijnsite.nl/stel-dit-deel-hier-in',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'sections',
        settings: {
          helpText: 'Kies de secties die met elkaar deze pagina vormen',
        },
        widgetId: 'entryLinksEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'theme',
        settings: { helpText: 'Kies het thema voor deze pagina' },
        widgetId: 'dropdown',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'seoTitle',
        settings: {
          helpText:
            'Titel van de pagina, gebruikt door zoek machines zoals Google',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'seoDescription',
        settings: {
          helpText:
            'Omschrijving van de pagina, gebruikt door zoek machines zoals Google',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'seoImage',
        settings: {
          helpText:
            'Afbeelding bij de pagina, wordt meegegeven aan een link en weergegeven door social media',
        },
        widgetId: 'assetLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'seoKeywords',
        settings: { helpText: '' },
        widgetId: 'tagEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'modelVersion',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
    ],
  },
  globals: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:21.169Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:26.403Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'globals', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
      },
      {
        fieldId: 'siteAuthor',
        settings: { helpText: 'Beschiknaar in alle secties' },
        widgetId: 'singleLine',
      },
      {
        fieldId: 'siteDescription',
        settings: {
          helpText:
            'Omschrijving van de site, gebruikt door zoek machines zoals Google',
        },
        widgetId: 'singleLine',
      },
      {
        fieldId: 'siteHeading',
        settings: {
          helpText:
            'Titel van de site, gebruikt door zoek machines zoals Google',
        },
        widgetId: 'singleLine',
      },
      {
        fieldId: 'siteKeywords',
        settings: {
          helpText:
            'Sleutelwoorden voor de site, gebruikt door zoek machines zoals Google',
        },
        widgetId: 'tagEditor',
      },
      {
        fieldId: 'siteImage',
        settings: { helpText: '' },
        widgetId: 'assetLinkEditor',
      },
      {
        fieldId: 'siteEmbeds',
        settings: { helpText: '' },
        widgetId: 'entryLinksEditor',
      },
      {
        fieldId: 'skipToContentHeading',
        settings: {
          helpText:
            'Tekst voor schermlezers om direct door te gaan naar de content.',
        },
        widgetId: 'singleLine',
      },
      { fieldId: 'modelVersion' },
    ],
  },
  embed: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:20.813Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:25.412Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'embed', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
      },
      { fieldId: 'code', settings: { helpText: '' }, widgetId: 'multipleLine' },
      { fieldId: 'type', settings: { helpText: '' }, widgetId: 'dropdown' },
      { fieldId: 'modelVersion' },
    ],
  },
  action: {
    sys: {
      id: 'default',
      type: 'EditorInterface',
      space: { sys: { id: 'tctsmza39jg6', type: 'Link', linkType: 'Space' } },
      version: 2,
      createdAt: '2021-08-05T21:24:20.506Z',
      createdBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      updatedAt: '2021-08-05T21:24:26.401Z',
      updatedBy: {
        sys: { id: '5TOHdotj5KMHTHKvucd6zZ', type: 'Link', linkType: 'User' },
      },
      contentType: {
        sys: { id: 'action', type: 'Link', linkType: 'ContentType' },
      },
      environment: {
        sys: { id: 'master', type: 'Link', linkType: 'Environment' },
      },
    },
    controls: [
      {
        fieldId: 'title',
        settings: {
          helpText:
            'Alleen gebruikt in Contentful, wordt nooit getoond op de site zelf',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'heading',
        settings: { helpText: 'Naam van de actie, zonder punt aan eind' },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'description',
        settings: { helpText: 'Omschrijving van de actie, voor zoekmachines' },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'icon',
        settings: { helpText: '' },
        widgetId: 'dropdown',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'page',
        settings: {
          helpText: 'De pagina waar de actie naar linkt (of gebruik de URL)',
        },
        widgetId: 'entryLinkEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'url',
        settings: {
          helpText:
            'De URL waar de actie naar linkt (of gebruik de paginalink)',
        },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'anchor',
        settings: { helpText: 'Voorbeeld: mijn-anchor' },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'query',
        settings: { helpText: 'Voorbeeld: &x=1&y=2' },
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'eventId',
        settings: { helpText: 'Voorbeeld: mijn-event-id' },
        widgetId: 'slugEditor',
        widgetNamespace: 'builtin',
      },
      {
        fieldId: 'modelVersion',
        widgetId: 'singleLine',
        widgetNamespace: 'builtin',
      },
    ],
  },
}
