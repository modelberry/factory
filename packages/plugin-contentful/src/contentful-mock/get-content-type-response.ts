export const getContentTypeResponse = {
  sys: {
    space: { sys: { type: 'Link', linkType: 'Space', id: 'tctsmza39jg6' } },
    id: 'testTopic',
    type: 'ContentType',
    createdAt: '2021-08-06T16:53:44.571Z',
    updatedAt: '2021-08-08T19:28:59.973Z',
    environment: {
      sys: { id: 'master', type: 'Link', linkType: 'Environment' },
    },
    publishedVersion: 19,
    publishedAt: '2021-08-08T19:28:59.973Z',
    firstPublishedAt: '2021-08-06T16:53:45.059Z',
    createdBy: {
      sys: { type: 'Link', linkType: 'User', id: '5TOHdotj5KMHTHKvucd6zZ' },
    },
    updatedBy: {
      sys: { type: 'Link', linkType: 'User', id: '5TOHdotj5KMHTHKvucd6zZ' },
    },
    publishedCounter: 10,
    version: 20,
    publishedBy: {
      sys: { type: 'Link', linkType: 'User', id: '5TOHdotj5KMHTHKvucd6zZ' },
    },
  },
  displayField: 'heading',
  name: 'testTopic',
  description: 'Topic model, a heading, an abstract and a call to action',
  fields: [
    {
      id: 'abstract',
      name: 'Abstract',
      type: 'Symbol',
      localized: false,
      required: false,
      validations: [
        { size: { max: 155, min: 0 } },
        { size: { max: 155, min: 0 } },
        { size: { max: 155, min: 0 } },
        { size: { max: 155, min: 0 } },
        { size: { max: 155, min: 0 } },
      ],
      disabled: false,
      omitted: false,
    },
    {
      id: 'actions',
      name: 'Actions',
      type: 'Array',
      localized: false,
      required: false,
      validations: [],
      disabled: false,
      omitted: false,
      items: {
        type: 'Link',
        validations: [
          { linkContentType: ['testAction'] },
          { linkContentType: ['testAction'] },
          { linkContentType: ['testAction'] },
          { linkContentType: ['testAction'] },
          { linkContentType: ['testAction'] },
        ],
        linkType: 'Entry',
      },
    },
    {
      id: 'heading',
      name: 'Heading',
      type: 'Symbol',
      localized: false,
      required: false,
      validations: [],
      disabled: false,
      omitted: false,
    },
    {
      id: 'icon',
      name: 'Icon',
      type: 'Symbol',
      localized: false,
      required: false,
      validations: [],
      disabled: false,
      omitted: false,
    },
    {
      id: 'media',
      name: 'Media',
      type: 'Array',
      localized: false,
      required: false,
      validations: [],
      disabled: false,
      omitted: false,
      items: { type: 'Link', validations: [], linkType: 'Asset' },
    },
    {
      id: 'poster',
      name: 'Poster',
      type: 'Link',
      localized: false,
      required: false,
      validations: [],
      disabled: false,
      omitted: false,
      linkType: 'Asset',
    },
  ],
}
