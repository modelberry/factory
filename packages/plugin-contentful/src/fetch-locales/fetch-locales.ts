import { Options, logger } from '@modelberry/mbfactory'
import { Environment, Locale } from 'contentful-management/types'

export interface FetchLocales {
  contentfulEnvironment: Environment
  options: Options
}

export type FetchLocalesReponse = {
  locales: Locale[]
  defaultLocaleCode: string
  cliLocaleCode?: string
  badCliLocale?: boolean
}

export const fetchLocales = async ({
  contentfulEnvironment,
  options,
}: FetchLocales) => {
  const result = await contentfulEnvironment.getLocales()
  const locales = result.items
  const defaultLocale = locales.find((locale: any) => locale.default)
  logger.p(`- remote default locale: ${defaultLocale?.code}`)

  const response: FetchLocalesReponse = {
    locales,
    defaultLocaleCode: defaultLocale?.code || 'en-US',
  }
  if (options.locale) {
    const cliLocale = locales.find((locale) => locale.code === options.locale)
    if (cliLocale) {
      response.cliLocaleCode = options.locale
    } else {
      logger.error(`- could not find cli locale ${options.locale} at remote`)
      response.badCliLocale = true
    }
  }
  return response
}
