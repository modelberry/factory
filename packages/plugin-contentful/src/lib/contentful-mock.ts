import { Environment, Locale } from 'contentful-management/types'

const entryMock = {
  mockedField: 'test',
}
const localeMock = {
  items: [
    {
      code: 'en-US',
      default: true,
      fallbackCode: 'en-US',
      name: 'English (United States)',
      optional: false,
    },
  ] as Locale[],
}
export const contentTypeMock = {
  fields: [],
  getEditorInterface: async () => contentTypeMock,
  publish: async () => contentTypeMock,
  update: async () => contentTypeMock,
}
export const createEntryWithId = jest.fn(async () => entryMock)
export const getLocales = jest.fn(async () => localeMock)
export const getContentType = jest.fn(async () => contentTypeMock)
export const createContentTypeWithId = jest.fn(async () => contentTypeMock)
export const environmentMock = {
  createEntryWithId,
  createContentTypeWithId,
  getContentType,
  getLocales,
} as unknown as Environment
export const getEnvironment = jest.fn(async () => environmentMock)
export const getSpace = jest.fn(async () => ({ getEnvironment }))
export const createClient = jest.fn(() => ({ getSpace }))
