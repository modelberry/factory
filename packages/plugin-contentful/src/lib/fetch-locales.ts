import chalk from 'chalk'
import { Environment } from 'contentful-management/types'

export interface FetchLocales {
  contentfulEnvironment: Environment
}

export const fetchLocales = async ({ contentfulEnvironment }: FetchLocales) => {
  const result = await contentfulEnvironment.getLocales()
  const locales = result.items
  const defaultLocale = locales.find((locale: any) => locale.default)
  console.log(chalk(`- remote default locale: ${defaultLocale?.code}`))
  return { locales, defaultLocale }
}
