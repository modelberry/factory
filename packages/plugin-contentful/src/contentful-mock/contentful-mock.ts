import { Environment, Locale } from 'contentful-management/types'
import { editorInterfaces } from './editor-interfaces'
import { getContentTypeResponse } from './get-content-type-response'
import { getContentTypesResponse } from './get-content-types-response'
import { getEntriesQueryResponse } from './get-entries-query-response'
import { getEntriesResponse } from './get-entries-response'

const entryMock = {
  fields: {},
  mockedField: 'test',
  publish: async () => entryMock,
  update: async () => entryMock,
}
const localeMock = {
  items: [
    {
      code: 'en-US',
      default: true,
      fallbackCode: 'en-US',
      name: 'English (United States)',
      optional: false,
    },
  ] as Locale[],
}
export const getContentTypeMock = async (contentType: any) => ({
  ...contentType,
  getEditorInterface: async () => ({
    ...editorInterfaces[contentType.name],
    update: async () => editorInterfaces[contentType.name],
  }),
  publish: async () => await getContentTypeMock(contentType),
  update: async () => await getContentTypeMock(contentType),
})

export const getContentTypesMock = async () => {
  for (const ct of getContentTypesResponse.items)
    Object.assign(ct, await getContentTypeMock(ct))
  return getContentTypesResponse
}

export const getEntriesMock = async (query: any) => {
  const entries = query.content_type
    ? getEntriesQueryResponse
    : getEntriesResponse
  for (const entry of entries.items)
    Object.assign(entry, await getContentTypeMock(entry))
  return entries
}

export const createEntryWithId = jest.fn(async () => entryMock)
export const createEntry = jest.fn(async () => entryMock)
export const getEntry = jest.fn(async () => entryMock)
export const getEntries = jest.fn(getEntriesMock)
export const getLocales = jest.fn(async () => localeMock)
export const getContentType = jest.fn(async () =>
  getContentTypeMock(getContentTypeResponse)
)
export const getContentTypes = jest.fn(getContentTypesMock)
export const createContentTypeWithId = jest.fn(async () =>
  getContentTypeMock(getContentTypeResponse)
)
export const environmentMock = {
  createEntry,
  createEntryWithId,
  createContentTypeWithId,
  getEntry,
  getEntries,
  getContentType,
  getContentTypes,
  getLocales,
} as unknown as Environment
export const getEnvironment = jest.fn(async () => environmentMock)
export const getSpace = jest.fn(async () => ({ getEnvironment }))
export const createClient = jest.fn(() => ({ getSpace }))
