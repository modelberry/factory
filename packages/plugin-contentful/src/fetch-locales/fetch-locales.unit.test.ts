import { logger } from '@modelberry/mbfactory'
import { environmentMock } from '../contentful-mock/contentful-mock'
import { fetchLocales } from './fetch-locales'

const logSpy = {
  h1: jest.spyOn(logger, 'h1').mockImplementation(),
  h2: jest.spyOn(logger, 'h2').mockImplementation(),
  h3: jest.spyOn(logger, 'h3').mockImplementation(),
  p: jest.spyOn(logger, 'p').mockImplementation(),
  info: jest.spyOn(logger, 'info').mockImplementation(),
  error: jest.spyOn(logger, 'error').mockImplementation(),
}

describe('getContentfulLocales should', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  test('get locales from mock', async () => {
    const locales = await fetchLocales({
      contentfulEnvironment: environmentMock,
      options: { locale: 'en-US' },
    })
    expect(locales).toMatchSnapshot()
    expect(logSpy.p).toHaveBeenCalledTimes(1)
  })
  test('return badCliLocate for a locale that does not exist from mock', async () => {
    const locales = await fetchLocales({
      contentfulEnvironment: environmentMock,
      options: { locale: 'es-ES' },
    })
    expect(locales).toMatchSnapshot()
  })
})
