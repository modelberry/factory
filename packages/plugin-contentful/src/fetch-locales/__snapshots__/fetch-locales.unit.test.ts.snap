// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getContentfulLocales should get locales from mock 1`] = `
Object {
  "cliLocaleCode": "en-US",
  "defaultLocaleCode": "en-US",
  "locales": Array [
    Object {
      "code": "en-US",
      "default": true,
      "fallbackCode": "en-US",
      "name": "English (United States)",
      "optional": false,
    },
  ],
}
`;

exports[`getContentfulLocales should return badCliLocate for a locale that does not exist from mock 1`] = `
Object {
  "badCliLocale": true,
  "defaultLocaleCode": "en-US",
  "locales": Array [
    Object {
      "code": "en-US",
      "default": true,
      "fallbackCode": "en-US",
      "name": "English (United States)",
      "optional": false,
    },
  ],
}
`;
